packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,54 +16,45 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    is_open,
+    toggleDrawer,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+}) => {
+    const [will_open, set_is_open] = useStateCallback(is_open);
 
-    componentWillReceiveProps(nextProps) {
-        this.setState({
-            is_open: nextProps.is_open,
-        });
+    const update_is_open = (value) => {
+        set_is_open(value)
     }
 
-    toggleDrawer = () => {
-        this.setState(
-            prev_state => ({
-                is_open: !prev_state.is_open,
-            }),
-            () => {
-                if (this.props.toggleDrawer) {
-                    this.props.toggleDrawer(this.state.is_open);
-                }
+    const toggle_drawer = () => {
+        update_is_open(!will_open)
+            return () => {
+                if (toggleDrawer) {
+                    toggleDrawer(will_open);
+                } 
```

Comment:
```suggestion
        setIsOpen(!is_open, () => {
            if (toggleDrawer) {
                 toggleDrawer(will_open);
            } 
        });
```
Using the `useStateCallback` you can pass a callback function as the second argument. This second argument will be fired as soon the state is set just like how `this.setState` work

Reply:
fixed, thanks 🙏